app/src/main/java/org/fossasia/openevent/activities/LocationActivtiy.java

```diff
@@ -0,0 +1,111 @@
+import android.support.v7.widget.SearchView;
+import android.support.v7.widget.Toolbar;
+import android.util.Log;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.adapters.SessionsListAdapter;
+import org.fossasia.openevent.data.Microlocation;
+import org.fossasia.openevent.data.Session;
+import org.fossasia.openevent.dbutils.DbSingleton;
+import org.fossasia.openevent.utils.IntentStrings;
+
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by MananWason on 8/18/2015.
+ */
+public class LocationActivtiy extends AppCompatActivity implements SearchView.OnQueryTextListener {
+    SessionsListAdapter sessionsListAdapter;
+    private Microlocation selectedLocation;
+    private List<Session> mSessions;
+    private RecyclerView sessionRecyclerView;
+    private String location;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_locations);
+        DbSingleton dbSingleton = DbSingleton.getInstance();
+        location = getIntent().getStringExtra(IntentStrings.MICROLOCATIONS);
+        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar_locations);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+
+        try {
+            selectedLocation = dbSingleton.getLocationByLocationname(location);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+
+        sessionRecyclerView = (RecyclerView) findViewById(R.id.recyclerView_locations);
+        mSessions = dbSingleton.getSessionbyLocationName(location);
+        sessionsListAdapter = new SessionsListAdapter(mSessions);
+        sessionRecyclerView.setLayoutManager(new LinearLayoutManager(this));
+        sessionRecyclerView.setAdapter(sessionsListAdapter);
+        sessionRecyclerView.setItemAnimator(new DefaultItemAnimator());
+    }
+
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
+
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_speakers_activity, menu);
+        SearchView searchView =
+                (SearchView) menu.findItem(R.id.search_sessions).getActionView();
+        searchView.setOnQueryTextListener(this);
+        return true;
+    }
+
+    @Override
+    public boolean onQueryTextSubmit(String query) {
+        return false;
+    }
+
+    @Override
+    public boolean onQueryTextChange(String query) {
+        DbSingleton dbSingleton = DbSingleton.getInstance();
+
+        mSessions = dbSingleton.getSessionbyLocationName(location);
```

Comment:
I am first making sure that I get the latest sessions lists for the database and then feeding the query after this.